pkg/providers/cloudstack/validator.go

```diff
@@ -244,6 +245,14 @@ func (v *Validator) validateMachineConfig(ctx context.Context, datacenterConfig
 		if err = v.cmk.ValidateServiceOfferingPresent(ctx, zone.Id, machineConfig.Spec.ComputeOffering); err != nil {
 			return fmt.Errorf("validating service offering: %v", err)
 		}
+		if err, fieldName, fieldValue := machineConfig.Spec.DiskOffering.Validate(); err != nil {
```

Comment:
this validation should be moved to config manager validation I mentioned before since it's context unaware.

Reply:
The validation in this commit is static validation which means it does not need extra info to do the validation and by design it should be under cloudstackmachineconfig_types.go's validate method. That validate method is triggered by config manager's reflection call so that it can be applied to every provider (vsphere/cloudstack, etc) that implements validate interface.

The original code is a TODO comment https://github.com/aws/eks-anywhere/blob/main/pkg/providers/cloudstack/validator.go#L124

I think this refactor is out of scope for this PR, and I am creating an issue to address diskoffering validate, service offering, compute offering and user validation. It seems at least vsphere and cloudstack needs these kind of refactor.